url4aac7e6f55_simon0191_tetris3d_tgz-pJ8-Tetris3dJ8/src/vsdk/toolkit/io/image/ImagePersistence.java

```diff
@@ -0,0 +1,693 @@
+    {
+        String type = extractExtensionFromFile(inImageFd);
+        RGBAImage retImage = new RGBAImage();
+
+        //- Try optimized reading, if native library is available ---------
+        if ( type.equals("png") ) {
+            nativeWarningGiven = true;
+        }
+
+        //-----------------------------------------------------------------
+        if( type.equals("tga") ) {
+            TargaImage t = new TargaImage(inImageFd);
+            t.exportRGBA(retImage);
+            return retImage;
+        }
+        else if( type.equals("jpg") || type.equals("jpeg") ||
+                 type.equals("gif") || type.equals("png") )  {
+            BufferedImage bi = null;
+
+            // OLD SLOW METHOD, DO NOT USE!
+            //java.awt.Toolkit awtTools = java.awt.Toolkit.getDefaultToolkit();
+            //java.awt.Image image;
+            //image = awtTools.getImage(inImageFd.getAbsolutePath());
+            //bi = toBufferedImage(image);
+
+            try {
+                BufferedInputStream bis;
+                FileInputStream fis;
+
+                fis = new FileInputStream(inImageFd);
+                bis = new BufferedInputStream(fis);
+                bi = ImageIO.read(bis);
+                bis.close();
+                fis.close();
+              }
+              catch ( Exception e ) {
+                  VSDK.reportMessage(null, VSDK.ERROR, "importRGBA",
+                                     "Cannot import image file \"" + inImageFd.getAbsolutePath() + "\"");
+                return null;
+            }
+            AwtRGBAImageRenderer.importFromAwtBufferedImage(bi, retImage);
+
+            return retImage;
+        }
+        throw new ImageNotRecognizedException("Image not recognized", inImageFd);
+    }
+
+    /**
+    Given a text string, this method determines if is a Unix style single
+    line comment, that is, if line begins with a '#' character.
+    */
+    private static boolean
+    isTextComment(String line)
+    {
+        char arr[] = line.toCharArray();
+        int i;
+
+        for ( i = 0; i < arr.length && (arr[i] != ' ' && arr[i] != '\t'); i++ );
+
+        if ( i < arr.length && arr[i] == '#' ) {
+            return true;
+        }
+        return false;
+    }
+
+    /**
+    Given the filename of an input data file which contains an image, this
+    method tries to recognize the file format and load the contents of it
+    to the image.
+
+    @todo Do not assume the file format only from the filename extension,
+    but trying to detect file headers.
+
+    @param imagen - The file respesenting the image
+    @return An RGBImage entity that contains the image loaded in memory.
+
+    Will change:
+      - Choose a better name for this method
+      - Do not recieve a File, but a Stream of bytes
+    */
+    public static RGBImage importRGB(File inImageFd) throws ImageNotRecognizedException, Exception
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

The cyclomatic complexity of this method is 17. By comparison, 99% of the methods in the CodeGuru reference dataset have a lower cyclomatic complexity. This indicates the method has a high number of decisions and it can make the logic difficult to understand and test. We recommend that you simplify this method or break it into multiple methods.